### PR TITLE
Update `KeyBindingContainer` usage to block key repeat where applicable

### PR DESCRIPTION
**Pull request by peppy**
- [x] Depends on ppy/osu-framework#4901.
- [x] Framework bump

Cases where repeat is still allowed:
- Changing game zoom
- Toggling toolbar
- Adjusting volume
- Seeking in editor
- Nudging in editor
- Editor clipboard functionality
- Toolbar buttons, skin editor toggle
- Song select footer buttons, random
- Gameplay menu overlay
- Gameplay chat display
- Drawable lounge item traversal

### CODE REVIEW

**Review by smoogipoo — DISMISSED** [review]
Seems good, still think areas like toolbar/song select footer allowing repeats is weird, but we'll see if anyone else says the same thing.

File: osu.Game/Screens/Ranking/ResultsScreen.cs [hunk 1]
```diff
@@ -330,6 +330,9 @@ private void onStatisticsStateChanged(ValueChangedEvent<Visibility> state)
 
         public bool OnPressed(KeyBindingPressEvent<GlobalAction> e)
         {
+            if (e.Repeat)
```
Comment (Joehuu) [inline osu.Game/Screens/Ranking/ResultsScreen.cs:333]: I think this should repeat, though the transition is abrupt right now.
Screen Recording 2021-11-17 at 9 37 37 PM

**Review by Joehuu — APPROVED** [review]
Looks good to me apart from one thing.

File: osu.Game/Screens/Ranking/ResultsScreen.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Screens/Ranking/ResultsScreen.cs:333]: why?

**Review by smoogipoo — APPROVED** [review]

File: osu.Game/Screens/Ranking/ResultsScreen.cs (hunk 1 above)
Comment (Joehuu) [reply osu.Game/Screens/Ranking/ResultsScreen.cs:333]: I slightly agree with the toolbar/overlay toggling to repeat and this is a similar case. Sure, fixing the animation is out of scope here, but if I can make it look less broken in a follow up PR, should it be a consideration?

Comment (peppy) [reply osu.Game/Screens/Ranking/ResultsScreen.cs:333]: I think i'd sooner make this a non-toggle. Select actions are generally supposed to be one directional. This is an outlier.

**Pull request merged**